### PR TITLE
Feat/result page

### PR DESCRIPTION
**Pull request by BalajiHarinath**
Added result page and authentication. Please review.

[Live link](https://splendid-muffin-5c9ba4.netlify.app/)

### CODE REVIEW

File: src/context/AuthContext.js [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+
+const AuthProvider = ({ children }) => {
+  const [authState, authDispatch] = useReducer(AuthReducer, InitialAuthData);
+  const { authData, authErrorMsg } = authState;
+
+  useEffect(() => {
+    const timerID = setTimeout(() => {
+      authDispatch({ type: "REMOVE_ERROR_MSG", payload: "" });
+    }, 3000);
+    return () => clearTimeout(timerID);
```
Comment (Dev-AkshatGupta) [inline src/context/AuthContext.js:15]: nice that you are using a unsubscribe function in the useEffect

File: src/components/RequireAuth/RestrictAuth.jsx [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import { Navigate, Outlet, useLocation } from "react-router-dom";
+import { useAuth } from "../../context";
+
+export const RestrictAuth = () => {
+  const { authData } = useAuth();
+  const userId = authData._id;
+  const location = useLocation();
+  console.log(location?.state);
+  console.log(location?.state?.from?.pathname);
```
Comment (Dev-AkshatGupta) [inline src/components/RequireAuth/RestrictAuth.jsx]: Remember to remove the consoles

File: src/pages/Authentication/authentication.css [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+  margin: auto;
+}
+
+.form-login h3 {
+  color: var(--primary-color);
+  margin: auto;
+}
+
+.form-field {
+  /* padding: 8px; */
```
Comment (Dev-AkshatGupta) [inline src/pages/Authentication/authentication.css]: Remove comments

**Review by Dev-AkshatGupta — APPROVED** [review]
Just have a look on the suggestions every thing else is fine 
You are good to go

File: src/pages/Result/Result.css [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+
+.prize-img {
+  height: fit-content;
+  width: 8rem;
+  object-fit: cover;
+}
+
+.result-container {
+  border: 1px solid var(--border-color);
+  border-radius: 1rem;
```
Comment (deekshithmd) [inline src/pages/Result/Result.css:14]: It is better to give border radius in px or else as the screen size changes radius also changes and will show unexpected shape.

File: src/pages/Result/Result.css [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+.option-container {
+  width: 100%;
+}
+
+.option-label-result {
+  width: inherit;
+  padding: 0.7rem 0.5rem;
+  background-color: var(--primary-color-light);
+  text-align: center;
+  border-radius: 0.5rem;
```
Comment (deekshithmd) [inline src/pages/Result/Result.css:27]: same here using px will be better

File: src/pages/Result/Result.jsx [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+            onClick={finishHandler}
+            className="finish-btn btn-solid btn-medium flex flex-justify-center flex-align-center"
+          >
+            Finish
+          </Link>
+        </div>
+      )}
+    </main>
+  );
+};
```
Comment (deekshithmd) [inline src/pages/Result/Result.jsx:119]: Look into those suggestions, rest everything looks good and you can proceed to merge

**Review by deekshithmd — APPROVED** [review]

**Comment by BalajiHarinath** [comment]
I have made the changes suggested. Thanks for the review

**Pull request merged**